### PR TITLE
OP-1064 Make list of exams laboratory parametric and pageable

### PR DESCRIPTION
**Pull request by ArnaudFonzam**
See OP-1064.

Paired with https://github.com/informatici/openhospital-core/pull/946.

### CODE REVIEW

File: src/main/java/org/isf/lab/rest/LaboratoryController.java [hunk 1]
```diff
@@ -347,15 +347,16 @@ public ResponseEntity<Boolean> deleteExam(@PathVariable Integer code) throws OHS
...
+	 * @param oneWeek
 	 * @return the {@link List} of found {@link LabWithRows} or NO_CONTENT otherwise.
 	 * @throws OHServiceException
 	 */
 	@GetMapping(value = "/laboratories", produces = MediaType.APPLICATION_JSON_VALUE)
-	public ResponseEntity<List<LabWithRowsDTO>> getLaboratory() throws OHServiceException {
+	public ResponseEntity<List<LabWithRowsDTO>> getLaboratory(@RequestParam boolean oneWeek, @RequestParam int pageNo, @RequestParam int pageSize) throws OHServiceException {
 		LOGGER.info("Get all LabWithRows");
-		List<Laboratory> labList = laboratoryManager.getLaboratory();
+		List<Laboratory> labList = laboratoryManager.getLaboratory(oneWeek, pageNo, pageSize);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/lab/rest/LaboratoryController.java:359]: in the Core PR, the param `oneWeek` is often (always?) written as `onWeek`.    Please pick a consistent name; thanks

File: src/main/java/org/isf/lab/rest/LaboratoryController.java [hunk 2]
```diff
@@ -347,15 +347,18 @@ public ResponseEntity<Boolean> deleteExam(@PathVariable Integer code) throws OHS
...
+	 * @param pageSize
 	 * @return the {@link List} of found {@link LabWithRows} or NO_CONTENT otherwise.
 	 * @throws OHServiceException
 	 */
 	@GetMapping(value = "/laboratories", produces = MediaType.APPLICATION_JSON_VALUE)
-	public ResponseEntity<List<LabWithRowsDTO>> getLaboratory() throws OHServiceException {
+	public ResponseEntity<List<LabWithRowsDTO>> getLaboratory(@RequestParam boolean onWeek, @RequestParam int pageNo, @RequestParam int pageSize) throws OHServiceException {
 		LOGGER.info("Get all LabWithRows");
-		List<Laboratory> labList = laboratoryManager.getLaboratory();
+		List<Laboratory> labList = laboratoryManager.getLaboratory(onWeek, pageNo, pageSize);
```
Comment (mwithi) [inline src/main/java/org/isf/lab/rest/LaboratoryController.java:361]: Shouldn't be there a pull-request also in core for this?

Comment (ArnaudFonzam) [reply src/main/java/org/isf/lab/rest/LaboratoryController.java:361]: yes

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

File: src/main/java/org/isf/lab/rest/LaboratoryController.java [hunk 3]
```diff
@@ -347,15 +347,18 @@ public ResponseEntity<Boolean> deleteExam(@PathVariable Integer code) throws OHS
...
 	 * 
+	 * @param oneWeek
+	 * @param pageNo
+	 * @param pageSize
 	 * @return the {@link List} of found {@link LabWithRows} or NO_CONTENT otherwise.
 	 * @throws OHServiceException
 	 */
 	@GetMapping(value = "/laboratories", produces = MediaType.APPLICATION_JSON_VALUE)
-	public ResponseEntity<List<LabWithRowsDTO>> getLaboratory() throws OHServiceException {
+	public ResponseEntity<List<LabWithRowsDTO>> getLaboratory(@RequestParam boolean onWeek, @RequestParam int pageNo, @RequestParam int pageSize) throws OHServiceException {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/lab/rest/LaboratoryController.java:359]: Why does the Javadoc have `oneWeek` and the method signature has `onWeek`?  Pick on or the other.

**Review by dbmalkovsky — APPROVED** [review]
Looks good except for one small naming issue.

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**